src/shmem_node_util.c

```diff
@@ -0,0 +1,92 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "shmem_internal.h"
+#include "shmem_node_util.h"
+
+
+char *shmem_internal_location_array = NULL;
+
+static int node_util_is_initialized = 0;
+
+
+int shmem_node_util_init(void) {
+    int ret;
+
+    if (!node_util_is_initialized) {
+        shmem_internal_location_array = malloc(sizeof(char) * shmem_internal_num_pes);
```

Comment:
Leaked during finalize?

Reply:
Whoops, yup.  Looks like it has always leaked...